### PR TITLE
Implemented unit tests for rescheduler

### LINKED ISSUE #2465: Rescheduler needs unit tests
This has been postponed over and over. Making P0.

cc: @piosz

### PR DESCRIPTION
**Pull request by piosz**
fix #2465

### CODE REVIEW

File: rescheduler/Makefile [hunk 1]
```diff
@@ -13,7 +13,7 @@ build: clean deps
 	$(ENVVAR) godep go build -o rescheduler
 
 test-unit: clean deps build
-	$(ENVVAR) godep go test --test.short -race ./... $(FLAGS)
```
Comment (mwielgus) [inline rescheduler/Makefile]: why without race checking?

**Comment by mwielgus** [comment]
What is the test coverage?

File: rescheduler/Makefile (hunk 1 above)
Comment (piosz) [inline rescheduler/Makefile]: done

**Comment by piosz** [comment]
@mwielgus it's not 100%. The test verify the most important functions.

**Comment by mwielgus** [comment]
Number please :)

**Comment by piosz** [comment]
35.4%
could be better

**Comment by piosz** [comment]
PTAL
coverage: 52.3% of statements

**Comment by fejta** [comment]
How are you getting coverage data? Teach me your magic! ;-)

**Comment by mwielgus** [comment]
There is `-coverage` option to `go test`.

**Comment by piosz** [comment]
Thanks @mwielgus !

**Pull request merged**